Todo/TaskListsViewController.swift

```diff
@@ -0,0 +1,271 @@
+                reloadIncompleteTasksCounts(incompTasksCountsLiveQuery.rows)
+            }
+    }
+
+    // MARK: - UITableViewController
+
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return listRows?.count ?? 0
+    }
+
+    override func tableView(tableView: UITableView,
+        cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+            let cell = tableView.dequeueReusableCellWithIdentifier("TaskListCell") as UITableViewCell!
+            
+            let doc = listRows![indexPath.row].document!
+            cell.textLabel?.text = doc["name"] as? String
+            
+            let incompleteCount = incompTasksCounts?[doc.documentID] ?? 0
+            cell.detailTextLabel?.text = incompleteCount > 0 ? "\(incompleteCount)" : ""
+            
+            return cell
+    }
+    
+    override func tableView(tableView: UITableView,
+        editActionsForRowAtIndexPath indexPath: NSIndexPath) -> [UITableViewRowAction]? {
+            let delete = UITableViewRowAction(style: .Normal, title: "Delete") {
+                (action, indexPath) -> Void in
+                // Dismiss row actions:
+                tableView.setEditing(false, animated: true)
+                // Delete list document:
+                let doc = self.listRows![indexPath.row].document!
+                self.deleteTaskList(doc)
+            }
+            delete.backgroundColor = UIColor(red: 1.0, green: 0.23, blue: 0.19, alpha: 1.0)
+            
+            let update = UITableViewRowAction(style: .Normal, title: "Edit") {
+                (action, indexPath) -> Void in
+                // Dismiss row actions:
+                tableView.setEditing(false, animated: true)
+                // Display update list dialog:
+                let doc = self.listRows![indexPath.row].document!
+                
+                Ui.showTextInputDialog(
+                    onController: self,
+                    withTitle: "Edit List",
+                    withMessage:  nil,
+                    withTextFieldConfig: { (textField) -> Void in
+                        textField.placeholder = "List name"
+                        textField.text = doc["name"] as? String
+                        textField.autocapitalizationType = .Words
+                    },
+                    onOk: { (name) -> Void in
+                        self.updateTaskList(doc, withName: name)
+                    }
+                )
+            }
+            update.backgroundColor = UIColor(red: 0.0, green: 0.48, blue: 1.0, alpha: 1.0)
+            
+            return [delete, update]
+    }
+    
+    // MARK: - UISearchController
+    
+    func updateSearchResultsForSearchController(searchController: UISearchController) {
+        let text = searchController.searchBar.text ?? ""
+        let active = !text.isEmpty
+        listsLiveQuery.startKey = active ? text : nil
```

Comment:
I think this code would be clearer if you used `if text.isEmpty { ... } else { ...}` instead of using two `?:`s.